eventsocket/eventsocket.go

```diff
@@ -336,7 +338,7 @@ func (h *Connection) Send(command string) (*Event, error) {
 		err error
 	)
 	select {
-	case err = <-h.err:
+	case err = <-h.ret:
 		return nil, err
```

Comment:
Does this also need patch? Not sure what'd be the err returned here.

Reply:
I think it should be like this: h.reet returns OK or ERROR messages for command execution, h.rerr returns things like network disconnection, protocol packet format error, etc. (critical error).

As said before: When I execute ExecuteUUID it should return an error message :"invalid session id [6e9d0022-3742-439e-a84d-78425dd7c376]", but since it's the same chan h.err, This err is read by ReadEvent, resulting in the final one minute timeoutPeriod returned. Originally, ExecuteUUID should be returned immediately, but because h.err was consumed by ReadEvent, the timeout returned.
	
select {
	case err = <-h.err:
		return nil, err      // should return from here
	case ev = <-h.cmd:
		return ev, nil
	case <-time.After(timeoutPeriod):
		return nil, errTimeout   // But it actually got stuck here for a minute before it timed out
}

Reply:
ReadEvent is called sequentially in a goroutine
	
	for {
		ev, err := c.ReadEvent()
		if err != nil {
			log.Fatal(err)
		}
		// do something
	}

Reply:
Call ExecuteUUID on the other goroutine.    Thank you for your reply